Explorer/Assets/DCL/SDKComponents/CameraControl/MainCamera/Tests/MainCameraSystemShould.cs

```diff
@@ -0,0 +1,337 @@
+                new PBVirtualCamera()
+                {
+                    DefaultTransition = new CameraTransition() { Speed = 30 }
+                });
+            entitiesMap[vCamCRDTEntity] = virtualCameraEntity1;
+
+            sdkCinemachineCam2 = new GameObject("SDKVirtualCamera1").AddComponent<CinemachineFreeLook>();
+            sdkCinemachineCam2.enabled = false;
+            sdkCinemachineCam1.transform.position = Vector3.one * -5.3f;
+            vCamComponent = new VirtualCameraComponent(sdkCinemachineCam2, -1);
+            vCamCRDTEntity = new CRDTEntity(223);
+            virtualCameraEntity2 = world.Create(vCamCRDTEntity, vCamComponent, new TransformComponent(sdkCinemachineCam2.transform),
+                new PBVirtualCamera()
+                {
+                    DefaultTransition = new CameraTransition() { Time = 2.5f }
+                });
+            entitiesMap[vCamCRDTEntity] = virtualCameraEntity2;
+
+            globalWorld = World.Create();
+            globalWorldCameraEntity = globalWorld.Create(
+                new CRDTEntity(SpecialEntitiesID.CAMERA_ENTITY),
+                new CameraComponent { Mode = CameraMode.ThirdPerson }
+            );
+            var cameraEntityProxy = new ObjectProxy<Entity>();
+            cameraEntityProxy.SetObject(globalWorldCameraEntity);
+
+            sceneStateProvider = Substitute.For<ISceneStateProvider>();
+            sceneStateProvider.IsCurrent.Returns(true);
+
+            cinemachineBrain = new GameObject("CinemachineBrain").AddComponent<CinemachineBrain>();
+            defaultCinemachineCam = new GameObject("DefaultCinemachineCam").AddComponent<CinemachineFreeLook>();
+            defaultCinemachineCam.enabled = true;
+            cinemachineBrain.ManualUpdate();
+            Assert.AreSame(defaultCinemachineCam.gameObject, cinemachineBrain.ActiveVirtualCamera.VirtualCameraGameObject);
+
+            cameraData = Substitute.For<IExposedCameraData>();
+            cameraData.CinemachineBrain.Returns(cinemachineBrain);
+            cameraData.CameraEntityProxy.Returns(cameraEntityProxy);
+
+            system = new MainCameraSystem(world, mainCameraEntity, entitiesMap, sceneStateProvider, cameraData, globalWorld);
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            entitiesMap.Clear();
+            world.Dispose();
+            globalWorld.Dispose();
+            GameObject.DestroyImmediate(sdkCinemachineCam1.gameObject);
+            GameObject.DestroyImmediate(sdkCinemachineCam2.gameObject);
+            GameObject.DestroyImmediate(defaultCinemachineCam.gameObject);
+            GameObject.DestroyImmediate(cinemachineBrain.gameObject);
+        }
+
+        [Test]
+        public void SetupMainCameraComponentCorrectly()
+        {
+            world.Remove<MainCameraComponent>(mainCameraEntity);
+
+            // Do not set up if not current scene
```

Comment:
I think each of this comments in the tests should be different tests since all of them are different cases. For example:
- DoNotAddCameraComponentWhenSceneIsNotCurrent
- AddCameraComponentWhenIsCurrentScene
- DoNotAddCameraComponentWhenIsNotInTheCameraEntity

Reply:
Good Idea, I separated that one into 4 smaller tests, along with some other tests.
[commit #1](https://github.com/decentraland/unity-explorer/pull/2036/commits/cd6f5cb6ce20f40f6f4f310078e2ce8d11f27ff2) + [commit #2](https://github.com/decentraland/unity-explorer/pull/2036/commits/e8162018e1ee54d583b1dfbaf6803525077c0380)